light_curves/code_src/sample_selection.py

```diff
@@ -318,7 +318,7 @@ def get_graham_sample(coords, labels, verbose=1):
 
 #SDSS QSO sample of any desired number
 #These are "normal" QSOs to use in the classifier
-def get_SDSS_sample(coords, labels, num,zmin=0,zmax=1, verbose=1):
+def get_sdss_sample(coords, labels, *, num=10, zmin=0, zmax=10, verbose=1):
```

Comment:
Did you intentionally change this to zmax = 10 from 1?

Reply:
Yes. Should it stay at 1? I was concerned that `num=500_000` would return fewer than 500K sources, and the user may not realize they need to increase `zmax`. I didn't actually try it. (I am skeptical of these z limits being here in the first place. I think they were committed by accident.)

Reply:
I think its good to have z_min, z_max in the sample build_up where possible. I might have added those in. And here's how the query can change to not sort by redshift by default I think:

query = f"SELECT TOP {num} specObjID, ra, dec, z FROM SpecObj \
WHERE ( z > 0.1 AND z < {zmax} AND class='QSO' AND zWARNING=0 ) \
ORDER BY NEWID()"

Reply:
good point about the 500k needing potentially a larger redshift range.  However, if I want a smaller sample, I want to know for instance that the lower redshift ones are high enough to not be extended sources, and the higher redshift ones are observable with some reasonable telescope to do follow up, ie., not z = 10.  Can we keep these variables visible so it is really obvious to users (including me!) what exactly is getting selected?  I think having them more visible would be a good nudge on the shoulder to not forget these other aspects of sample selection.

Reply:
What do you mean by "more visible"?

Reply:
ahhhh, maybe that is the difference between me and someone who writes better code than me....I think function definitions in a different file than the main notebook are one level removed, and therefore 'hidden' or 'less visible' to the user.  I assume most users won't actually look at the function definitions.  Maybe you don't start with that assumption?

Reply:
@xoubish do you want me to add that option to randomize z? If so, do you want the randomization on or off by default?

Randomizing z will result in different samples being returned each time (at the very least, they will be in a different order). This means the sample is not reproducible. That's my biggest hesitation, so I would leave it off by default.

Reply:
@jkrick would I agree that code in a different file than the notebook is "less visible" and I don't expect most users to look at it. Are you just wanting the z arguments to be shown in the main notebook where this gets called?

Reply:
Why would ordering by ID be randomized and not reproducible?

Reply:
```
ORDER BY NEWID()
```

As I understand it, this creates a new, random ID every time. So ```SELECT TOP N``` will return different rows every time.

Reply:
You are right. And apparently it doesn't get a seed. I am not sure if people want to build sample more than once that they care about getting the same sample. It doesn't bother me much to go with sorted redshifts either. I will use this on the forked version for publication but here it can go either way.

Reply:
> I will use this on the forked version for publication but here it can go either way.

Ah, ok then I'm happy to add it here. There are some things that will need to be different in your fork (otherwise you wouldn't need a fork), but functionality like this doesn't need to be one of them. The less out-of-sync your fork is with this repo, the easier it will be to manage.

Reply:
I've added both a `randomize_z` option in this function and a demo of `z` kwargs in the main notebook.